[JBPM-10032] Support file vault credentials provider

ruromero opened the pull request:
Signed-off-by: ruromero <rromerom@redhat.com>

**JIRA**: 

[JBPM-10032](https://issues.redhat.com/browse/JBPM-10032)

Support for [Quarkus File Vault](https://github.com/quarkiverse/quarkus-file-vault) allowing defining the credentials-provider property for any password in the application configuration. Includes:

* datasources
* kieserver authentication
* client-certs

ruromero committed:
[JBPM-10032] Support file vault credentials provider

Signed-off-by: ruromero <rromerom@redhat.com>

ruromero commented:
I just updated the commit message to align the JBPM jira issue

afalhambra commented on src/main/java/org/kie/processmigration/model/config/KieClientCert.java:
```diff
@@ -18,6 +18,8 @@
 
 import java.util.Optional;
 
+import javax.swing.text.html.Option;
```
Not used, please remove

afalhambra commented on src/main/java/org/kie/processmigration/service/impl/KieServiceImpl.java:
```diff
@@ -76,7 +80,10 @@
+    private static final String CREDENTIALS_PROVIDER_USER_KEY = "user";
+    private static final String CREDENTIALS_PROVIDER_PASSWORD_KEY = "password";
 
+    private CredentialsProvider credentialsProvider = CredentialsProviderFinder.find("quarkus.file.vault");
```
maybe this can be final and static

ruromero replied:
No because it uses injection and if it were static the ArcContainer won't have been initialized

afalhambra commented on src/main/java/org/kie/processmigration/service/impl/KieServiceImpl.java:
```diff
@@ -76,7 +80,10 @@
     private static final long AWAIT_EXECUTOR = 5;
     private static final long RETRY_DELAY = 2;
     private static final Logger logger = LoggerFactory.getLogger(KieServiceImpl.class);
+    private static final String CREDENTIALS_PROVIDER_USER_KEY = "user";
+    private static final String CREDENTIALS_PROVIDER_PASSWORD_KEY = "password";
```
Why not just use the out-of-the-box constant? `io.quarkus.credentials.CredentialsProvider.PASSWORD_PROPERTY_NAME` and `io.quarkus.credentials.CredentialsProvider.PASSWORD_PROPERTY_NAME`?

ruromero replied:
nice catch, thanks

afalhambra commented on examples/kieservers/basic-auth-vault.yml:
```diff
@@ -0,0 +1,10 @@
+  file:
+    vault:
+      provider:
+        pim:
```
not sure about this, but as per `credentials-provider` property defined above, shouldn't we need to include another entry within `pim` containing `kieserver`?:

```suggestion
        pim:
           kieserver:
```

ruromero replied:
I've used `kieserver1` key instead.

afalhambra commented on examples/kieservers/client-cert-vault.yml:
```diff
@@ -0,0 +1,9 @@
+  - host: http://localhost:18080/kie-server/services/rest/server
+client-cert:
+    cert-name: pim
+    cert-credentials-provider: quarkus.file.vault.provider.pim.cert
+    keystore-path: /path/to/keystore.jks
+    keystore-credentials-provider: quarkus.file.vault.provider.pim.keystore
+    truststore-path: /path/to/truststore.jks
+    truststore-credentials-provider: quarkus.file.vault.provider.pim.truststore
```
It would be nice and clearer if you could also add the properties in yaml for the `quarkus.file.vault.provider` entries

afalhambra commented on examples/vault/vault.yml:
```diff
@@ -0,0 +1,9 @@
+  file:
+    vault:
+      provider:
+        pim:
```
same as basic-auht-yaml file, shouldn't we need to add another entry matching with the below `credentials-provider`? such as `quarkus.file.vault.provider.pim.dbpwd`?

afalhambra reviewed (commented):
Just some minor comments, otherwise it looks great. 
Nice job @ruromero

afalhambra commented:
@ruromero - forgot to mention, that it would nice to do the same example of configuration for a DB using vault, as it's part of the jira request

ruromero committed:
[JBPM-10032] Support for file vault
Add examples and keys to the vault
Removed unuse dimport
Use provided constants in the CredentialsProvider

Signed-off-by: ruromero <rromerom@redhat.com>

ruromero commented:
@afalhambra changes applied, thanks for the review

afalhambra reviewed (approved):
outstanding job

ruromero committed:
[JBPM-10032] upgrade to quarkus-file-vault 0.2.0

Signed-off-by: ruromero <rromerom@redhat.com>

elguardian commented:
@ruromero needs to be backported to 7.67.x
